src/main/scala/ch/openolitor/buchhaltung/BuchhaltungCommandHandler.scala

```diff
@@ -306,6 +334,21 @@ trait BuchhaltungCommandHandler extends CommandHandler with BuchhaltungDBMapping
       RechnungsPositionAssignToRechnung(rechnungId, index)
     )
   }
+
+  private def checkTemplateInvoice(body: String, subject: String, ids: Seq[RechnungId])(implicit session: DBSession): Boolean = {
+    val templateCorrect = ids flatMap { rechnungId: RechnungId =>
+      buchhaltungReadRepository.getPerson(rechnungId) flatMap { person =>
+        buchhaltungReadRepository.getById(rechnungMapping, rechnungId) map { rechnung =>
+          val mailContext = RechnungMailContext(person, rechnung)
+          generateMail(subject, body, mailContext) match {
+            case Success(mailPayload) => true
+            case Failure(e)           => false
```

Comment:
replace with:
generateMail(...).isDefined

Reply:
There is a isSuccess

Reply:
lovely